src/env/obfuscated.rs

```diff
@@ -89,6 +90,30 @@ impl FileSystem for ObfuscatedFileSystem {
     type Handle = <DefaultFileSystem as FileSystem>::Handle;
     type Reader = ObfuscatedReader;
     type Writer = ObfuscatedWriter;
+    type AsyncIoContext = <DefaultFileSystem as FileSystem>::AsyncIoContext;
+
+    fn async_read(
+        &self,
+        ctx: &mut Self::AsyncIoContext,
+        handle: Arc<Self::Handle>,
+        block: &FileBlockHandle,
+    ) -> IoResult<()> {
+        self.inner.async_read(ctx, handle, block)
+    }
+
+    fn async_finish(&self, ctx: &mut Self::AsyncIoContext) -> IoResult<Vec<Vec<u8>>> {
+        let base = self.inner.async_finish(ctx).unwrap();
+        let mut res = vec![];
+        for v in base {
+            let mut temp = vec![];
```

Comment:
I don't know why you need to create another vector instead of modifying `base` directly.